app/services/OvhService.php

```diff
@@ -0,0 +1,117 @@
+    static function emailsSettingsValidation($user, $v_infos, $real_email, $nose_email)
+    {
+        $toast_message = "";
+        try {
+            // Check if user is part of a family
+            if ($user->family->id && $user->family_leader) {
+                // Get all family members with the same email
+                $family_members = em()->getRepository(User::class)->findBy([
+                    'family' => $user->family,
+                    'real_email' => $user->real_email
+                ]);
+                // Update emails and redirection for every family member
+                foreach ($family_members as $family_member) {
+                    ovh_api()->removeRedirection($family_member->nose_email, $family_member->real_email);
+                    $family_member->real_email = $real_email->value;
+                    ovh_api()->addRedirection($family_member->nose_email, $real_email->value);
+                    em()->persist($family_member);
+                }
+                $toast_message = count($family_members) > 1 ? "Emails mis à jour pour toute la famille !" : "Emails mis à jour !";
+            } else {
+                // Update emails and redirection for the user only
+                ovh_api()->removeRedirection($user->nose_email, $user->real_email);
+                $user->real_email = $real_email->value;
+                $user->nose_email = $nose_email->value;
+                ovh_api()->addRedirection($nose_email->value, $real_email->value);
+                em()->persist($user);
+                $toast_message = "Emails mis à jour !";
+            }
+        } catch (GuzzleHttp\Exception\ClientException $e) {
+            logger()->error("Erreur lors de l'ajout de la redirection", ["exception" => $e]);
+            $v_infos->set_error("Erreur de redirection");
+            return false;
+        }
+        em()->flush();
+        Toast::create($toast_message);
+        return true;
+    }
+
+    static function userDeactivateValidation($form, $user)
+    {
+        try {
+            $mailingLists = ovh_api()->getMailingLists();
+            foreach ($mailingLists as $list) {
+                if (in_array($user->nose_email, ovh_api()->getMailingListSubscribers($list))) {
+                    ovh_api()->removeSubscriberFromMailingList($list, $user->nose_email);
+                    logger()->info("User {$user->id} removed from mailing list {$list})");
+                }
+            }
+            ovh_api()->removeRedirection($user->nose_email, $user->real_email);
+            logger()->info("User {$user->id} got his redirection {$user->nose_email} -> {$user->real_email} removed");
+        } catch (GuzzleHttp\Exception\ClientException $e) {
+            logger()->error("Erreur dans la mise à jour des redirections ou des listes d'email", ["exception" => $e]);
+            $form->set_error("Erreur dans la mise à jour des redirections ou des listes d'email");
+            return false;
```

Comment:
if we are going to do this, then we need bulletproof logging. UserId for example is mandatory here, maybe also knowing which api call failed (we have this from the exception probably)